xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/internal/transformation/macro/DefaultMacroTransformationConfiguration.java

```diff
@@ -43,12 +43,20 @@ public class DefaultMacroTransformationConfiguration implements MacroTransformat
      */
     private Properties macroCategories = new Properties();
 
+    private Set<String> hiddenCategories = Set.of();
+
     @Override
     public Properties getCategories()
     {
         return this.macroCategories;
     }
 
+    @Override
+    public Set<String> getHiddenCategories()
+    {
+        return this.hiddenCategories;
```

Comment:
Isn't there a problem with the terminology? If Hidden categories are categories then surely getCategories should return them? I don't recall our discussion re hidden macros. If we decided to implement them has categories then I don't understand why we need a new API. Thx

Reply:
ah I remember now, sorry. It's just that we can configure which categories are hidden! :)

Reply:
The way the API is currently used, it's easier to have to separate Sets, but indeed that's something that could be useful in the future.